content/general/concepts/what-is-functional-programming/what-is-functional-programming.md

```diff
@@ -0,0 +1,53 @@
+---
+Title: 'Functional Programming'
+Description: 'Encourages programming with pure functions and evaluating expressions instead of statements.'
+Subjects:
+  - 'Code foundations'
+  - 'Computer Science'
+  - 'Web Development'
+Tags:
+  - 'Functions'
+  - 'Recursion'
+CatalogContent:
+  - 'learn-to-code'
+  - 'paths/code-foundations'
+---
+
+_Functional Programming_ (FP) is a declarative software development paradigm which encourages constructing programs by composing pure functions and evaluating expressions instead of statements. The functions are isolated and independent of the state of the application.
```

Comment:
```suggestion
Functional Programming (FP) is a declarative software development paradigm that encourages constructing programs by composing pure functions and evaluating expressions instead of statements. The functions are isolated and independent of the state of the application.
```